scripts/generate_sitemap.py

```diff
@@ -111,45 +123,60 @@ def generate(self):
+            except Exception as e:
+                self.log_errors(obj, obj._id, e)
 
         # Registration urls
         for obj in Registration.objects.filter(retraction=False, is_deleted=False, is_public=True).iterator():
-            config = settings.SITEMAP_REGISTRATION_CONFIG
-            config['loc'] = urlparse.urljoin(settings.DOMAIN, obj.url)
-            config['lastmod'] = obj.date_modified.isoformat()
-            self.add_url(config)
+            try:
+                config = settings.SITEMAP_REGISTRATION_CONFIG
+                config['loc'] = urlparse.urljoin(settings.DOMAIN, obj.url)
+                config['lastmod'] = obj.date_modified.isoformat()
+                self.add_url(config)
+            except Exception as e:
+                self.log_errors(obj, obj._id, e)
 
         # Preprint urls
         for obj in PreprintService.objects.filter(node__isnull=False, node__is_deleted=False, node__is_public=True, is_published=True).iterator():
-            preprint_date = obj.date_modified.isoformat()
-            config = settings.SITEMAP_PREPRINT_CONFIG
-            config['loc'] = urlparse.urljoin(settings.DOMAIN, obj.url)
-            config['lastmod'] = preprint_date
-            self.add_url(config)
-
-            # Preprint file urls
-            file_config = settings.SITEMAP_PREPRINT_FILE_CONFIG
-            file_config['loc'] = urlparse.urljoin(settings.DOMAIN, 
-                os.path.join('project', 
-                    obj.primary_file.node._id, # Parent node id
-                    'files',
-                    'osfstorage',
-                    obj.primary_file._id, # Preprint file deep_url
-                    '?action=download'
-                )
-            )
-            file_config['lastmod'] = preprint_date
-            self.add_url(file_config)
+            try:
+                preprint_date = obj.date_modified.isoformat()
+                config = settings.SITEMAP_PREPRINT_CONFIG
+                config['loc'] = urlparse.urljoin(settings.DOMAIN, obj.url)
+                config['lastmod'] = preprint_date
+                self.add_url(config)
+
+                # Preprint file urls
+                try:
+                    file_config = settings.SITEMAP_PREPRINT_FILE_CONFIG
+                    file_config['loc'] = urlparse.urljoin(settings.DOMAIN, 
+                        os.path.join('project', 
+                            obj.primary_file.node._id, # Parent node id
+                            'files',
+                            'osfstorage',
+                            obj.primary_file._id, # Preprint file deep_url
+                            '?action=download'
+                        )
+                    )
+                    file_config['lastmod'] = preprint_date
+                    self.add_url(file_config)
+                except Exception as e:
+                    self.log_errors(obj.primary_file, obj.primary_file._id, e)
+            except Exception as e:
```

Comment:
This might be a little grabby, but it's probably fine. I'm expecting the first few rounds of work post-release will be tightening this up a bit.